addons/base_import/models.py

```diff
@@ -261,6 +263,112 @@ def _read_csv(self, record, options):
             if any(x for x in row if x.strip())
         )
 
+    def _try_match_column(self, cr, uid, index, preview_values, fields, options):
```

Comment:
Maybe this whole thing should be a loop on tentative detection methods for clarity, trying them one after another and returning when one matches?